### ISSUE TITLE
Error during training 

### ISSUE THREAD

**Issue by Gouiaa** [ildoonet/tf-pose-estimation#383]
Hi @ildoonet 
I am training the `mobilenetthin` model and I get this error when the training is finished.

2019-01-22 14:30:56.479582: W tensorflow/core/kernels/queue_base.cc:277] _0_fifo_queue: Skipping cancelled enqueue attempt with queue not closed
[2019-01-22 14:30:56,479] [train] [INFO] optimization finished. 169.224656
[2019-01-22 14:30:56,481] [pose_dataset] [ERROR] err type1, placeholders=[<tf.Tensor 'image:0' shape=(4, 368, 368, 3) dtype=float32>, <tf.Tensor 'heatmap:0' shape=(4, 46, 46, 19) dtype=float32>, <tf.Tensor 'vectmap:0' shape=(4, 46, 46, 38) dtype=float32>]
[2019-01-22 14:30:56,482] [pose_dataset] [INFO] Thread-1 Exited.

**Comment by luckhhn** [comment]
Hello, I am a newcomer, and I also want to ask you about the training. I also want to train the mobilenetthin model from scratch, and then I don't know which data set should be downloaded. Here are 2014,2016 and 2017

Did you download it under the link below?
2017 Train images [118K/18GB]
2017 Val images [5K/1GB]
2017 Test images [41K/6GB]

Does its tag information download this?
2017 Train/Val annotations [241MB]
2017 Stuff Train/Val annotations [1.1GB]
2017 Panoptic Train/Val annotations [821MB]
2017 Testing Image info [1MB]

And then, how do you organize these data sets and labels?Can you give me a detailed demonstration?

**Comment by luckhhn** [comment]
@Gouiaa  hello, Looking forward to your reply

**Comment by meetdave06** [comment]
@luckhhn  Yes, the links are correct. Download 2017 one. For keypoint annotations download 2017 Train/Val annotations. Keep both of them in a folder. You'll have to give the annotation and image path while training

**Comment by meetdave06** [comment]
@luckhhn  I am working on Ubuntu 18.04 environment so I don't have much idea about win10. But looking at the error log, it seems that you don't have cudnn installed. Kindly set it up and try again.

**Comment by luckhhn** [comment]
@meetdave06  thanks， mask.py err= No module named 'pycocotools._mask'  is frist error

 Could not find 'cudnn64_7.dll'. TensorFlow requires that this DLL be installed in a directory that is named in your %PATH% environment variable. Note that installing cuDNN is a separate step from installing CUDA, and this DLL is often found in a different directory from the CUDA DLLs. You may install the necessary DLL by downloading cuDNN 7 from this URL: https://developer.nvidia.com/cudnn    ，this is second error

I have successfully installed gpu-tensorflow,tensorflow= 1.10.cudu = 9.0, I can run this demo successfully, now I want to train myself, but there is an error, I am very confused.I would like to ask you the results of your training?Then can you tell me the specific GPU configuration?This includes the adjustment of your training parameters, as well as your training environment, such as tensorflow version and cudn version

thank you very much！

**Comment by meetdave06** [comment]
@luckhhn  The first error won't stop your training. You only have cuda installed. You need to set up cudnn as well. 

My config:- Tensorflow=1.8, Ubuntu 18.04 , CUDA 9.0, CUDNN 7.4.2
  
I am running it on AWS instance which has Tesla K 80 GPU. It is not sufficient for training and is quite slow.

**Comment by luckhhn** [comment]
@meetdave06 Thank you for your reply, no, I installed cudnn, cuda=9.0, cudnn=9,0. I have already run the demo provided by the author, that is, there is no environmental problem.So I don't understand, I'm confused, and it makes me stop training from a perspective, which is sad

**Comment by luckhhn** [comment]
@meetdave06 How did your training turn out?Any results?Batch = 16?Learning rate = 0.001?

**Comment by luckhhn** [comment]
mask.py err= No module named 'pycocotools._mask'  The first error won't stop your training？But I checked that I didn't configure the coco dataset API

**Comment by meetdave06** [comment]
@luckhhn There is no cudnn 9.0. The latest one is 7.4.2. I stopped the training after an hour as it was very slow and stuck on 'Training started'. I used batch=16 and learning rate = 0.0001. I am going to try it on a better GPU

**Comment by luckhhn** [comment]
@meetdave06   【tensorflow】windows10 +Anaconda3+cuda9.0+cudnn9.0+tensorflow1.10.0+Opencv3.4 https://blog.csdn.net/qq_37791134/article/details/81605721    I use the cudnn-9.0-windows10-x64-v7

If you have the training result, I will wait for your notification. I want to know the effect of the training. Thank you

**Comment by meetdave06** [comment]
@luckhhn Sure, I will let you know once the training is completed.

**Comment by luckhhn** [comment]
@Gouiaa thanks ，I am a Chinese student. Now I am in the third year of graduate school. I will graduate soon.

**Comment by luckhhn** [comment]
@meetdave06 hello，I met the following two difficult problems. Could you tell me what you did to train successfully?I downloaded coco2017. I changed the path and followed the training steps. Is there anything I should pay attention to?
1： RuntimeError: can't start new thread
2：BlockingIOError: [Errno 11] Resource temporarily unavailable

**Comment by finnickniu** [comment]
@Gouiaa Hi, could you tell me whether you solve the issue please

**Comment by aspenlin** [comment]
Hi @Gouiaa, I'm also training on AWS with 1 GPU. And from my estimation it will take me about 10 days to train cmu model for only 2 epochs with 16 batchsize and 0.001 learning rate. From your experience is this normal? I did notice that my neither my GPU nor my memory is fully utilized. What hardware did you switch to and how much time is it taking you to train? Any help is appreciated.

**Comment by Gouiaa** [comment]
Hi @aspenlin,
I have worked a little bit on this code but it was really mess and not clear at all. It is better to start implementing the code from scratch here is a very useful repo to start [https://github.com/michalfaber/keras_Realtime_Multi-Person_Pose_Estimation](url)

**Comment by BadMachine** [comment]
@rafikg did u solve your problem from your post? Looks like total mess in this thread and @luckhhn started to asking irrelevant questions

**Comment by weilanShi** [comment]
> Hi @ildoonet
> I am training the `mobilenetthin` model and I get this error when the training is finished.
> 
> 2019-01-22 14:30:56.479582: W tensorflow/core/kernels/queue_base.cc:277] _0_fifo_queue: Skipping cancelled enqueue attempt with queue not closed
> [2019-01-22 14:30:56,479] [train] [INFO] optimization finished. 169.224656
> [2019-01-22 14:30:56,481] [pose_dataset] [ERROR] err type1, placeholders=[<tf.Tensor 'image:0' shape=(4, 368, 368, 3) dtype=float32>, <tf.Tensor 'heatmap:0' shape=(4, 46, 46, 19) dtype=float32>, <tf.Tensor 'vectmap:0' shape=(4, 46, 46, 38) dtype=float32>]
> [2019-01-22 14:30:56,482] [pose_dataset] [INFO] Thread-1 Exited.
can you provide the model of  mobilenetthin, thanks
